src/transformers/generation_flax_utils.py

```diff
@@ -283,6 +281,41 @@ def generate(
             # prepare decoder_input_ids for generation
             input_ids = jnp.ones((input_ids.shape[0], 1), dtype="i4") * decoder_start_token_id
 
+        # Prepare `max_length` depending on other stopping criteria.
+        input_ids_seq_length = input_ids.shape[-1]
+        if max_length is None and max_new_tokens is None:
+            warnings.warn(
```

Comment:
Very nice! I like it. It'll create a lot of warnings in the beginning, but there is no way around it